test/unit/cloudwatch_agent/test_write_cloudwatch_agent_json.py

```diff
@@ -0,0 +1,194 @@
+        "node_roles": ["ComputeFleet", "HeadNode"],
+        "platforms": ["amazon", "centos"],
+        "feature_conditions": [{"dna_key": "dcv_enabled", "satisfying_values": ["head_node"]}],
+    },
+    {
+        "timestamp_format_key": "month_first",
+        "file_path": "/var/log/syslog",
+        "log_stream_name": "syslog",
+        "log_group_name": "pre-existing",
+        "schedulers": ["awsbatch", "slurm"],
+        "node_roles": ["ComputeFleet", "HeadNode"],
+        "platforms": ["ubuntu"],
+        "feature_conditions": [{"dna_key": ["directory_service", "enabled"], "satisfying_values": ["true"]}],
+    },
+    {
+        "timestamp_format_key": "default",
+        "file_path": "/var/log/cfn-init.log",
+        "log_stream_name": "cfn-init",
+        "schedulers": ["awsbatch"],
+        "node_roles": ["HeadNode"],
+        "platforms": ["amazon", "centos", "ubuntu"],
+        "feature_conditions": [],
+    },
+]
+
+METRIC_CONFIGS = {
+    "metrics_collected": [
+        {
+            "metric_type": "mem",
+            "measurement": ["mem_used_percent"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+        {
+            "metric_type": "disk",
+            "measurement": ["disk_used_percent"],
+            "resources": ["/"],
+            "metrics_collection_interval": 60,
+            "node_roles": ["HeadNode"],
+        },
+    ],
+    "append_dimensions": ["InstanceId"],
+    "aggregation_dimensions": [["InstanceId", "path"]],
+}
+
+
+def test_add_log_group_name_params():
+    configs = add_log_group_name_params("test", CONFIGS)
+    for config in configs:
+        assert_that(config).contains("log_group_name")
+        assert_that(config["log_group_name"]).is_equal_to("test")
+
+
+def test_add_instance_log_stream_prefixes():
+    configs = add_instance_log_stream_prefixes(CONFIGS)
+    for config in configs:
+        assert_that(config["log_stream_name"]).contains(gethostname())
+
+
+@pytest.mark.parametrize(
+    "dimensions",
+    [
+        {"platform": "amazon", "length": 2},
+        {"platform": "ubuntu", "length": 2},
+        {"scheduler": "slurm", "length": 2},
+        {"scheduler": "awsbatch", "length": 3},
+    ],
+)
+def test_select_configs_for_dimesion(dimensions):
+    if "platform" in dimensions.keys():
+        configs = select_configs_for_platform(CONFIGS, dimensions["platform"])
+        assert_that(len(configs)).is_equal_to(dimensions["length"])
+    else:
+        configs = select_configs_for_scheduler(CONFIGS, dimensions["scheduler"])
+        assert_that(len(configs)).is_equal_to(dimensions["length"])
+
+
+@pytest.mark.parametrize(
+    "info",
+    [
+        {"node_info": {"dcv_enabled": "head_node", "directory_service": {"enabled": "true"}}, "length": 3},
+        {"node_info": {"directory_service": {"enabled": "true"}}, "length": 2},
+        {"node_info": {"enabled": "true"}, "length": 1},
+    ],
```

Comment:
You can keep this as to reduce using the usage of dict keys
 @pytest.mark.parametrize(
    "node_info, length",
    [
        ({"dcv_enabled": "head_node", "directory_service": {"enabled": "true"}}, "length": 3),
        ({"directory_service": {"enabled": "true"}}, "length": 2),
       ({"enabled": "true"}, "length": 1),
    ],